cognito_code_grant/signed_cookies.py

```
@@ -0,0 +1,132 @@
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import padding
+from botocore.signers import CloudFrontSigner
+
+# Heavily inspired on a couple of existing solutions in stackoverflow and AWS documentation.
+# Adapted for good measure, to serve our needs.
+
+def rsa_signer(message):
+    return private_key.sign(message, padding.PKCS1v15(), hashes.SHA1())
+
+
+def generate_signed_cookies(resource=None,expire_minutes=5):
+    """
+    @resource   path to s3 object inside bucket(or a wildcard path,e.g. '/blah/*' or  '*')
+    @expire_minutes     how many minutes before we expire these access credentials (within cookie)
+    return tuple of domain used in resource URL & dict of name=>value cookies
+    """
+    if not resource:
+        resource = '*'
+
+    return dist.create_signed_cookies(resource,expire_minutes=expire_minutes)
+
+
+class SignedCookiedCloudfrontDistribution():
+
+    def __init__(self,connection,download_dist_id,cname=True):
+        """
+        @download_dist_id   id of your Cloudfront download distribution
+        @cname          boolean True to use first domain cname, False to use 
+                        cloudfront domain name, defaults to cname
+                        which presumably matches your writeable cookies ( .mydomain.com)
+        """
+        self.download_dist = None
+        self.domain = None
+        try:
+            download_dist = connection.get_distribution(Id=download_dist_id)
+            if cname and download_dist.get('Distribution', {}).get('AliasICPRecordals', {}):
+                self.domain = download_dist['Distribution']['AliasICPRecordals'][0]['CNAME'] #use first cname if defined
+            else:
+                self.domain = download_dist.domain_name
+            self.download_dist = download_dist
+        except Exception as ex:
+            logging.error(ex)
```

Should we swallow this exception? Seems like if this fails, the later call to `get_http_resource_url` will just generate an invalid result (something like `http://None/...`).

---

I might remove this entire function and just pass the domain via env var. Gonna run a few tests and see if it suits our needs.